nucypher/policy/conditions/lingo.py

```diff
@@ -116,6 +120,24 @@ def eval(self, data) -> bool:
                 f"'{self.value}' is an unprocessed context variable and is not valid "
                 f"for condition evaluation."
             )
+        if is_context_variable(self.key):
+            # programming error if we get here
+            raise RuntimeError(
+                f"'{self.key}' is an unprocessed context variable and is not valid "
+                f"for condition evaluation."
+            )
+        if self.key and type(data) is dict:
```

Comment:
To keep things clean, maybe pull the processing of the key into a helper method.